News-Android-App/src/main/java/de/luhmer/owncloudnewsreader/SettingsFragment.java

```diff
@@ -236,17 +248,38 @@ private void bindGeneralPreferences(DialogPreference.TargetFragment prefFrag)
         bindPreferenceSummaryToValue(prefFrag.findPreference(SP_SWIPE_LEFT_ACTION));
     }
 
+    /**
+     * migrates pref SYNC_INTERVAL_IN_MINUTES_STRING to pref_sync_settings
+     * temporary function, could be removed whenever is wished
+     */
+    private void migrateSyncIntervalValue() {
```

Comment:
I recommend to create a new database version (sic!) which does not change the database, but can be used for a migration that affects shared preferences, too. This way it is decoupled from the logic of the current version.

I don't know how you usually handle shared preferences migrations in the news app @David-Development?

Reply:
Not sure how to handle this one.
Would you have any input to help progress on this ?

Reply:
Well, it's not my app, so i am not sure how @David-Development handles this kind of migrations (or if he wants to handle it at all).

In the Deck & Notes app i handle this by bumping the database version (without any actual change in the database) to utilize the hook for changing the shared preferences.

Example:

https://github.com/stefan-niedermann/nextcloud-notes/blob/master/app/src/main/java/it/niedermann/owncloud/notes/persistence/NotesDatabase.java#L78

and

https://github.com/stefan-niedermann/nextcloud-notes/blob/master/app/src/main/java/it/niedermann/owncloud/notes/persistence/migration/Migration_19_20.java#L28

(The concrete implementation for the News app will be slightly different because here `SQLiteOpenHelper` is used instead of the `Room` lib, but the concept can still be used for a clean migration.)

Reply:
We never had any migration of SharedPreferences so far so this is the first time this comes up. From my point of view the way @fabienli implemented is great! Thank you! 🙇